operators/advanced_network/python/check_system.py

```diff
@@ -0,0 +1,632 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import re
+import subprocess
+import sys
+
+
+def setup_logging():
+    """
+    Configures the logging settings.
+    """
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(asctime)s - %(levelname)s - %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+
+
+def parse_args():
+    """
+    Parses command-line arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description="Check system tuning for ANO performance",
+        epilog=(
+            "Examples:\n"
+            "  python check_system.py -cpu-freq       # Check CPU frequency governor\n"
+            "  python check_system.py -mrrs     # Check MRRS settings for NVIDIA NICs\n\n"
+            "  python check_system.py -mps      # Check max payload size settings for NVIDIA NICs\n\n"
+        ),
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    parser.add_argument(
+        "--check",
+        choices=[
+            "all",
+            "cpu-freq",
+            "mrrs",
+            "mps",
+            "hugepages",
+            "gpu-clocks",
+            "bar1-size",
+            "topo",
+            "cmdline",
+            "mtu",
+        ],
+        help=(
+            "Specify the property to check:\n"
+            "  all        - Perform all checks\n"
+            "  cpu-freq   - Check if the CPU frequency governor is set to 'performance'.\n"
+            "  mrrs       - Check if the Maximum Read Request Size (MRRS) of NVIDIA NICs is set to 4096.\n"
+            "  mps        - Check if the Maximum Payload Size is set to 256B.\n"
+            "  hugepages  - Check if hugepages are enabled\n"
+            "  gpu-clocks - Check GPU clocks\n"
+            "  bar1-size  - Check the BAR1 size of the GPU\n"
+            "  topo       - Check the GPU and NIC topology\n"
+            "  cmdline    - Check the kernel boot parameters\n"
+            "  mtu        - Check MTU of each NVIDIA interface\n"
+        ),
+    )
+
+    parser.add_argument("--set", choices=["mrrs"], help=("  mrrs      - Update MRRS of NICs\n"))
+    return parser.parse_args()
+
+
+def get_nic_info():
+    """
+    Parses the output of `ibdev2netdev -v` to extract and return a list of tuples,
```

Comment:
some systems don't have this (isn't that technically for RDMA?).

Could we instead do the same as [here](https://github.com/nvidia-holoscan/holohub/blob/1b8c0569f21b216685be18140e9b166cdbfa6c66/applications/adv_networking_bench/cpp/CMakeLists.txt#L130-L140) which doesn't depend on any tool installation?

```bash
bus_address_list=$(lspci | grep -E \"Ethernet.*ConnectX\" | awk '{print $1}')
...
interface_name=$(find /sys/class/net -exec bash -c 'echo {} $(readlink -f {}/device)' \; | grep ${bus_address} | awk '{print $1}')
```

Reply:
I think at this point we've resorted to ANO being for NVIDIA NICs only and it requires the ANO container. That container always contains this script. The script above works on today's card's (maybe?), but it may break on future generations. For example, does it work on BF?

Reply:
> we've resorted to ANO being for NVIDIA NICs only and it requires the ANO container. That container always contains this script

That's fair, though one would probably run the script outside the container just locally (which I did for testing), but then not an issue when running the app.

>  The script above works on today's card's (maybe?), but it may break on future generations. For example, does it work on BF?

The command for `interface_name` is going to work always. For the first one though, there probably can be a better heuristic to list the NICs. If we can figure that out that would be nice. `lspci | grep "Ethernet controller"` is safe also, but will grab any (example: the rj45 on the igx too).

That's fine not to block this MR since you pointed out `ibdev2netdev` is in the container. Would be great to think about a more generic way if we can, so lmk if you have an answer to the thoughts above.

Reply:
>The command for interface_name is going to work always.

The command is based on a grep of lspci where the name of the device doesn't change. It would be more correct to do something like look at the PCI vendor ID. We don't want to include other NIC types (like on-board NICs captured by "Ethernet Controller" since those don't have any of the capabilities we're checking for.

Reply:
Nice. Seems like the first command would then be
```bash
lspci -nn | grep -iE 'Ethernet controller.*\[15b3:'
```
Can keep that for another round